modules/fluid_properties/src/fluidproperties/SalineMoltenSaltFluidProperties.C

```diff
@@ -0,0 +1,133 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//
+#include "SalineMoltenSaltFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", SalineMoltenSaltFluidProperties);
+
+InputParameters
+SalineMoltenSaltFluidProperties::validParams()
+{
+  InputParameters params = SinglePhaseFluidProperties::validParams();
+  params.addClassDescription("Molten salt fluid properties using Saline");
+  params.addRequiredParam<std::vector<std::string>>("comp_name",
+                                                    "The name of the components in the salt");
+  params.addRequiredParam<std::vector<Real>`>("comp_val",
+                                              "The mole fraction of each salt component");
+  params.addRequiredParam<std::string>(
+      "prop_def",
+      "Definition of a fluid property file, which must be a file path to the "
+      "comma-separated data matching the Saline format.");
+  return params;
+}
+
+SalineMoltenSaltFluidProperties::SalineMoltenSaltFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters)
+{
+#ifdef SALINE_ENABLED
+  const auto & propDef = getParam<std::string>("prop_def");
+  _d.load(propDef);
+  bool success = _tp.initialize(&_d);
+  if (!success)
+    mooseError("The initialization of the Saline interface has failed");
+  const auto & name = getParam<std::string>("comp_name");
+  const auto & comp = getParam<std::string>("comp_val");
```

Comment:
That's not going to work. You need to update the code here and below to the new parameter type. Instead of tokenize you will now have to stringify to build the `-` separated string for saline.

Reply:
fixed